docs/spec/spec.md

```diff
@@ -0,0 +1,416 @@
+
+An IDoc can be sent using the following function:
+
+```ballerina
+# Send the iDoc
+#
+# + iDoc - The XML string of the iDoc
+# + iDocType - The type of the iDoc
+# + return - An error if the execution fails
+remote function sendIDoc(xml iDoc, IDocType iDocType = jco:DEFAULT) returns Error
+```
+
+The input parameters require `iDoc`, with an optional `iDocType`. The `iDoc` represents the content of the IDoc in XML format. The `iDocType` specifies the version of the IDoc being sent and can be set to `DEFAULT`, `VERSION_2`, `VERSION_3`, `VERSION_3_IN_QUEUE`, or `VERSION_3_IN_QUEUE_VIA_QRFC`. If no `iDocType` is provided, the `DEFAULT` type will be applied.
+
+Users can invoke it as shown below:
+
+```ballerina
+public function main() returns error? {
+    xml iDoc = check io:fileReadXml("resources/sample.xml");
+    check jcoClient->sendIDoc(iDoc);
+    io:println("IDoc sent successfully.");
+}
+```
+
+#### 2.1.3 Data Type Mapping Between Ballerina and SAP JCo
+
+The following table maps Ballerina data types to their corresponding SAP JCo types:
+
+| Ballerina Type                        | SAP JCo Type   | SAP Data Type         |
+| ------------------------------------- | -------------- | --------------------- |
+| string                                | char           | CHAR                  |
+| string                                | char[]         | STRING, SSTRING       |
+| time:Date                             | Date           | DATE                  |
+| time:TimeOfDay                        | Time           | TIME                  |
+| int                                   | short          | INT1                  |
+| int                                   | int            | INT2                  |
+| int                                   | long           | INT4                  |
+| float                                 | float          | FLOAT                 |
+| decimal                               | double         | DEC, QUAN, CURR, PREC |
+| byte                                  | byte           | INT1                  |
+| byte[]                                | byte[]         | RAW, LRAW, BYTE       |
+| decimal                               | BigDecimal     | DEC, QUAN, CURR       |
+| int                                   | BigInteger     | NUM                   |
+| record (with/without rest fields)     | JCoStructure   | STRUCTURE             |
+| record[] (with/without rest fields)   | JCoTable       | TABLE                 |
+
+### 2.2 Listener
+
+#### 2.2.1 Configurations
+
+The Listener supports two types of configurations, `ServerConfig` and `AdvancedConfig`, to facilitate communication with the SAP system by receiving IDocs and handling events.
+
+##### 2.2.1.1 ServerConfig
+
+The `ServerConfig` type represents the configuration details needed to create an IDoc connection.
+
+```ballerina
+public type ServerConfig record {
```

Comment:
Any reason to mark the `ServerConfig` as an open record ? Usual practice is to use closed records.

Reply:
Yes. Fixed with https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/57.